frontend/src/Hooks/useForm.js

```diff
@@ -0,0 +1,26 @@
+import { useState } from "react";
+
+const emailRegex = /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
```

Comment:
Instead of pasting unreadable regexes, check this out: https://github.com/jquense/yup - a very nice library for all kinds of validations (including emails)